pkg/apis/certificates/validation/validation.go

```diff
@@ -58,6 +60,52 @@ func ValidateCertificateSigningRequest(csr *certificates.CertificateSigningReque
 	if len(csr.Spec.Usages) == 0 {
 		allErrs = append(allErrs, field.Required(specPath.Child("usages"), "usages must be provided"))
 	}
+	if csr.Spec.SignerName == nil {
+		allErrs = append(allErrs, field.Required(specPath.Child("signerName"), "signerName must be provided"))
+	} else {
+		// ensure signerName is of the form domain.com/something and up to 571 characters
+		if len(*csr.Spec.SignerName) > 571 {
+			allErrs = append(allErrs, field.TooLong(specPath.Child("signerName"), *csr.Spec.SignerName, 571))
+		}
+		parts := strings.Split(*csr.Spec.SignerName, "/")
+		if len(parts) != 2 {
+			allErrs = append(allErrs, field.Invalid(specPath.Child("signerName"), *csr.Spec.SignerName, "must be a fully qualified domain of the form 'example.com/signer-name'"))
+		} else {
+			domain := parts[0]
+			path := parts[1]
+			signerName := *csr.Spec.SignerName
+
+			// ensure the path portion has a length greater than 0
+			if len(domain) == 0 {
+				allErrs = append(allErrs, field.Invalid(specPath.Child("signerName"), signerName, "domain component must be specified"))
+			}
+
+			// check the domain portion to ensure it consists of valid dns labels
+			domainOnDots := strings.Split(domain, ".")
```

Comment:
Do we need to ensure there is at least one `.`?

Reply:
I don't think that needs to be a hard requirement - I'd personally say `example/something` is a valid `signerName`, but I'm not sure what the standard in other similar fields is.

Reply:
+1 on using IsFullyQualifiedName or IsQualifiedname.

Reply:
I've opted for `IsFullyQualifiedDomainName` for now as we can always loosen this requirement in future if there's a need for it. I also opted to *not* used `IsFullyQualifiedName`as that method is deprecated (and actually requires *three* `.` to be valid) 😄